src/login/logind-user.c

```diff
@@ -344,6 +347,23 @@ int user_load(User *u) {
         return 0;
 }
 
+static void user_start_runtime_dir(User *u) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        int r;
+
+        assert(u);
+        assert(!u->stopping);
+        assert(u->manager);
+        assert(u->runtime_dir_service);
+
+        u->service_job = mfree(u->service_job);
```

Comment:
so, hmm, there's still only a single `service_job` field? is that a good idea?  can't it happen that we start waiting for too much, i.e. for the manager job to finish just to start the manager seession, thus causing a deadlock?

somehow I think we should stop being too smart here, and add distinct fields for the runtime dir job and the manager job.

Reply:
> can't it happen that we start waiting for too much, i.e. for the manager job to finish just to start the manager seession, thus causing a deadlock?

No, as in `session_ready` we check `service_job` only if `SESSION_CLASS_WANTS_SERVICE_MANAGER`. I'll add a few comments later.

-----------------

The reason why I really don't want that is that currently the job tracking logic is simple - we track one job only, and thus we don't need to care about individual job states before replacing it. I.e., when we start user@.service, it doesn't matter whether the previous job for user-runtime-dir@.service is a start or a stop job.

If we change this, however, we need to check the type of the job before we can clear it. If runtime-dir is stopping, and we now would start user@.service, and `session_ready` checks runtime_dir_job or service_job depending on the session class, then we would have a problem since we don't really know the state of user-runtime-dir@.service. And all those complexity don't seem to resolve any real problem.

(Actually I had this PR initially like that, but later found out this approach more manageable...)

Reply:
so the problem is that we actually want to allow through the light/manager sessions only after u-r-d@.s is complete. Hence, we *always* want to know the state of one of the two services for each session type. it just depends which one...